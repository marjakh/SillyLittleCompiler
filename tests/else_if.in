function f1(x) {
  write(11);
  if (x == 0) {
    write(0);
  } else if (x == 1) {
    write(1);
  }
}

f1(0);
f1(1);
f1(2);

function f2(x) {
  write(22);
  if (x == 0) {
    write(0);
  } else if (x == 1) {
    write(1);
  } else {
    write(2);
  }
}

f2(0);
f2(1);
f2(2);

function f3(x) {
  write(33);
  if (x == 0) {
    write(0);
  } else if (x == 1) {
    write(1);
  } else if (x == 2) {
    write(2);
  }
}

f3(0);
f3(1);
f3(2);
f3(3);

function f4(x) {
  write(44);
  if (x == 0) {
    write(0);
  } else if (x == 1) {
    write(1);
  } else if (x == 2) {
    write(2);
  } else {
    write(3);
  }
}

f4(0);
f4(1);
f4(2);
f4(3);
