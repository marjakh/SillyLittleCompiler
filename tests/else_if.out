11
0
11
1
11
22
0
22
1
22
2
33
0
33
1
33
2
33
44
0
44
1
44
2
44
3
